tools/Makefile

```diff
@@ -0,0 +1,65 @@
+CC ?= gcc
```

Comment:
Move the targets for generating tools in `mk/tools.mk` and integrate it into top-level `Makefile`.